app/lib/request.rb

```diff
@@ -111,16 +111,9 @@ def perform
     end
 
     begin
-      # If we are using a persistent connection, we have to
-      # read every response to be able to move forward at all.
-      # However, simply calling #to_s or #flush may not be safe,
-      # as the response body, if malicious, could be too big
-      # for our memory. So we use the #body_with_limit method
-      response.body_with_limit if http_client.persistent?
```

Comment:
Wouldn't this change require that every caller that uses persistent connection fully consume the response? Is that something we're sure we're currently doing?

Reply:
True. Even though we ignore the body of the POST response, we need to consume anything returned by the server.

I have  now changed `#perform` to always consume up to 1 MB. This is plenty for the regular case. Only if a server sends more than that, the persistent connection will be closed.